lib/minesweeper_2pl/cli.rb

```diff
@@ -1,4 +1,178 @@
-module Minesweeper_2pl
+module Minesweeper
   class CLI
+    def print(msg)
+      puts msg
+    end
+
+    def start
+      welcome
+      get_player_params
+    end
+
+    def get_move(game)
+      ask_for_move
+      get_player_input(game)
+    end
+
+    def get_player_params
+      result = []
+      size = nil
+      while size.nil?
+        ask_for_board_size
+        size = get_player_entered_board_size
+      end
+      result << size
+      count = nil
+      while count.nil?
+        ask_for_bomb_count(size)
+        count = get_player_entered_bomb_count(size * size)
+      end
+      result << count
+      result
+    end
+
+    def get_player_input(game)
+      input = gets.chomp
+      if player_input_has_correct_format(input)
+        return_coordinates_if_input_is_within_range(input, game)
+      else
+        puts invalid_player_input_message
+      end
+    end
+
+    def get_player_entered_board_size
+      input = gets.chomp
+      if board_size_input_has_correct_format(input)
+        return_row_size_if_input_is_within_range(input)
+      else
+        puts invalid_row_size_message
+      end
+    end
+
+    def get_player_entered_bomb_count(board_size)
+      input = gets.chomp
+      if bomb_count_input_has_correct_format(input)
+        return_bomb_count_if_input_is_within_range(input, board_size)
+      else
+        puts invalid_bomb_count_message
+      end
+    end
+
+    def welcome
+      puts "\n==========================================="
+      puts "           WELCOME TO MINESWEEPER          "
+      puts "===========================================\n\n"
+    end
+
+    def ask_for_move
+      string = "\nPlayer 1, make your move:\n- to place a move: enter the word 'move' followed by one digit from the header and one digit from the left column, eg. move 3,1:\n- to place (or remove) a flag: enter the word 'flag' followed by the desired coordinates eg flag 3,1\n"
+      print(string)
+    end
+
+    def ask_for_board_size
+      puts "Player 1 please enter a row size for your board, any number less than or equal to 20. \n(Entering 20 will give you a 20X20 board)\n"
+    end
+
+    def ask_for_bomb_count(size)
+      puts "Player 1 please enter the number of bombs there should be on the board. \n(The number should not be more than #{(size * size * App::BOMB_PERCENT).to_i})"
+    end
+
+    def show_game_over_message(result)
+      if result == "win"
+        puts "Game over! You win!"
+      elsif result == "lose"
+        puts "Game over! You lose."
+      end
+    end
+
+    def invalid_move
+      puts "That was not a valid move. Please try again."
+    end
+
+    private
+
+    def player_input_has_correct_format(input)
```

Comment:
There are a _lot_ of helper methods in the interface of this class. I wonder if a handful of simple `Validator` classes could help simplify it and hide the details about validating input from this class.